management/src/test/java/org/ehcache/management/registry/DefaultManagementRegistryServiceTest.java

```diff
@@ -138,9 +157,16 @@ public void descriptorOffHeapTest() {
       assertThat(new ArrayList<Capability>(managementRegistry.getCapabilities()).get(1).getName(), equalTo("StatisticsCapability"));
       assertThat(new ArrayList<Capability>(managementRegistry.getCapabilities()).get(2).getName(), equalTo("StatisticCollectorCapability"));
       assertThat(new ArrayList<Capability>(managementRegistry.getCapabilities()).get(3).getName(), equalTo("SettingsCapability"));
-
       assertThat(new ArrayList<Capability>(managementRegistry.getCapabilities()).get(0).getDescriptors(), hasSize(4));
-      assertThat(new ArrayList<Capability>(managementRegistry.getCapabilities()).get(1).getDescriptors(), hasSize(55));
+
+      Collection<Descriptor> descriptors = new ArrayList<Capability>(managementRegistry.getCapabilities()).get(1).getDescriptors();
+      Collection<Descriptor> allDescriptors = new ArrayList<Descriptor>();
+      allDescriptors.addAll(StatsUtil.ONHEAP_DESCRIPTORS);
+      allDescriptors.addAll(StatsUtil.OFFHEAP_DESCRIPTORS);
+      allDescriptors.addAll(StatsUtil.CACHE_DESCRIPTORS);
+
+      assertThat(descriptors, hasSize(allDescriptors.size()));
+      assertThat(descriptors, containsInAnyOrder(allDescriptors.toArray()));
```

Comment:
please make this assertion before the hasSize. Otherwise the error message we'll get will still be 50 vs 55 and not the list of missing stats